src/measure_theory/constructions/polish.lean

```diff
@@ -781,3 +784,54 @@ begin
 end
 
 end polish_space
+
+
+namespace measure_theory
+
+-- todo after the port: move to topology/metric_space/polish
+instance [polish_space α] : polish_space (univ : set α) := is_closed_univ.polish_space
+
+variables (α) [measurable_space α] [polish_space α] [borel_space α]
+
+lemma exists_nat_measurable_equiv_range_coe_fin_of_finite [finite α] :
+  ∃ n : ℕ, nonempty (α ≃ᵐ range (coe : fin n → ℝ)) :=
+begin
+  obtain ⟨n, ⟨n_equiv⟩⟩ := finite.exists_equiv_fin α,
+  refine ⟨n, ⟨polish_space.equiv.measurable_equiv (n_equiv.trans _)⟩⟩,
+  exact equiv.of_injective _ (nat.cast_injective.comp fin.val_injective),
+end
+
+lemma measurable_equiv_range_coe_nat_of_infinite_of_countable [infinite α] [countable α] :
+  nonempty (α ≃ᵐ range (coe : ℕ → ℝ)) :=
+begin
+  haveI : polish_space (range (coe : ℕ → ℝ)),
+  { exact nat.closed_embedding_coe_real.is_closed_map.closed_range.polish_space, },
+  refine ⟨polish_space.equiv.measurable_equiv _⟩,
+  refine (nonempty_equiv_of_countable.some : α ≃ ℕ).trans _,
+  exact equiv.of_injective coe nat.cast_injective,
+end
+
+/-- Any Polish Borel space is measurably equivalent to a subset of the reals. -/
+theorem exists_subset_real_measurable_equiv : ∃ s : set ℝ, measurable_set s ∧ nonempty (α ≃ᵐ s) :=
```

Comment:
Isn't it more natural to phrase this by saying that there exists a measurable embedding from `α` to `ℝ`? (Also for applications: then you can push forward a measure by this measurable embedding, and you won't need to discuss subtypes).

Reply:
Thanks for the suggestion! I added a lemma stating the existence of an embedding.